app.py

```diff
@@ -1,28 +1,67 @@
 from datetime import datetime
+from pprint import pprint
+import requests
 
 from flask import Flask, redirect, render_template, request, url_for
+from flask_login import LoginManager, login_required, login_user, current_user
+from oauthlib.oauth2 import WebApplicationClient
 
+from api.auth import authorisation_disabled, get_user, requires_role
+from api.auth_config import AuthConfig
 from api.mongo_config import MongoConfig
 from api.mongo_helper import MongoHelper
 import api.session_items as session
 import api.sorter as sorter
+from models.role import Role
+from models.user import User
 from models.view_model import ViewModel
 
 def create_app():
     app = Flask(__name__)
     config = MongoConfig()
     data_manager = MongoHelper(config)
+    app.secret_key = config.SECRET_KEY
     app.data_manager = data_manager
     app.config.from_object(config)
+    auth_config = AuthConfig()
+
+    client = WebApplicationClient(auth_config.AUTH_CLIENT_ID)
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        uri = client.prepare_request_uri('https://github.com/login/oauth/authorize')
+        return redirect(uri)
+
+    @app.route('/login/callback')
+    def login_callback():
+        code_param = request.args.get("code")
+        url, headers, body = client.prepare_token_request('https://github.com/login/oauth/access_token', client_id=auth_config.AUTH_CLIENT_ID, client_secret=auth_config.AUTH_CLIENT_SECRET, code=code_param)
+        token_response = requests.post(url, body, headers=headers)
+        access_token = client.parse_request_body_response(token_response.text)['access_token']
+        github_user_profile = requests.get("https://api.github.com/user", headers={"Authorization": f"Bearer {access_token}"}).json()
```

Comment:
Not important, but worth noting that the `parse_request_body` stores the relevant values with the client for later access to try make things easier, so these lines could equivalently look like..
```python
        client.parse_request_body_response(token_response.text)
        user_uri, headers, body = client.add_token("https://api.github.com/user")
        github_user_profile= requests.get(user_uri, headers=headers).json()
```